src/main/java/com/internet/shop/dao/impl/OrderDaoListImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Dao
+public class OrderDaoListImpl implements OrderDao {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getUserId());
+        order.setProducts(List.copyOf(shoppingCart.getProducts()));
+        shoppingCartDao.clear(shoppingCart);
+        Storage.addOrder(order);
+        return order;
+    }
```

Comment:
Let's move the 'complete' method to service layer and create a method 'create' instead.

Reply:
Also, we should clear the cart after completing the order and then update it.